credit_card.rb

```diff
@@ -35,21 +35,24 @@ def to_s
 
   # return a new CreditCard object given a serialized (JSON) representation
   def self.from_s(card_s)
-    # TODO: deserializing a CreditCard object
+    #   - input: a JSON string
+    #   - output: a CreditCard object
+    JSON.parse(card_s)
   end
 
   # return a hash of the serialized credit card object
   def hash
-    # TODO: implement this method
     #   - Produce a hash (using default hash method) of the credit card's
     #     serialized contents.
     #   - Credit cards with identical information should produce the same hash
+    to_json.hash
   end
 
   # return a cryptographically secure hash
   def hash_secure
-    # TODO: implement this method
     #   - Use sha256 to create a cryptographically secure hash.
     #   - Credit cards with identical information should produce the same hash
+    require 'digest'
```

Comment:
I think require statements should be at the top?

Reply:
Oh I didn't noticed that.
Fixed.